### PR TITLE
[Affliction] Added several legendaries

### PR DESCRIPTION
**Pull request by sMteX**
Added few of the legendary items for Affliction. I would wait till I have all of them but you said you prefer smaller PRs so there we go.

I am well aware of the code duplicity of Soul Harvest and The Master Harvester modules - I will try to do the same as with Death's Embrace and Soul of the Netherlord but unfortunately it's not as trivial as it looks like (trying to determine which effect triggered the buff, so I can accurately account the damage bonus to the talent or the chest - same as with Topless Tower and Avenging Wrath for Holy Paladin). I have some ideas, will try to implement in next PR.

### CODE REVIEW

File: src/Parser/AfflictionWarlock/Modules/Items/Legendaries/TheMasterHarvester.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class TheMasterHarvester extends Module {
+  playerBonusDmg = 0;
+  petBonusDmg = 0;
+
+  petIds = [];
+  on_initialized() {
+    if (!this.owner.error) {
+      this.active = this.owner.selectedCombatant.hasChest(ITEMS.THE_MASTER_HARVESTER.id);
+    }
+    this.owner.report.friendlyPets.filter(pet => pet.petOwner === this.owner.playerId).forEach(pet => {
```
Comment (MartijnHols) [inline src/Parser/AfflictionWarlock/Modules/Items/Legendaries/TheMasterHarvester.js]: This does not support mid-combat pet summons.

File: src/Parser/AfflictionWarlock/Modules/WarlockCore/DeathsEmbrace.js [hunk 2]
```diff
@@ -54,19 +49,6 @@ class DeathsEmbrace extends Module {
     }
     this.bonusDmg += getDamageBonus(event, this.getDeathEmbraceBonus(targetHealthPercentage));
   }
-
-  statistic() {
```
Comment (MartijnHols) [inline src/Parser/AfflictionWarlock/Modules/WarlockCore/DeathsEmbrace.js]: Why did you remove this statistic? It seemed useful.

**Comment by MartijnHols** [comment]
Thanks! Please review my comments.

File: src/CHANGELOG.js [hunk 3]
```diff
@@ -2,7 +2,6 @@ export default `
 20-08-2017 - Restoration Shaman: Added Ancestral Vigor metric. (by Yuyz0112)
 19-08-2017 - Guardian Druid: Added detail on Ironfur usage.
 19-08-2017 - Vengeance Demon Hunter: Added more useful and specific statistics tooltips.
-18-08-2017 - Added Affliction Warlock support by @Chizu.
```
Comment (sMteX) [inline src/CHANGELOG.js]: Don't really know, I updated my fork ("Merge remote-tracking branch 'upstream/master'") and when I looked through what was changed I noticed there is another line with Added Affliction Warlock support. Don't know where that came from but I liked the colored version more :D

File: src/Parser/AfflictionWarlock/Modules/Items/Legendaries/TheMasterHarvester.js (hunk 1 above)
Comment (sMteX) [inline src/Parser/AfflictionWarlock/Modules/Items/Legendaries/TheMasterHarvester.js]: That's probably right, I don't yet have a log that I could test this on, but my guess is that all pets summoned throughout the instance (since that's what is in this.report, no?) should be in the in the array, and re-summoned pets should have the same id. I would test this on training dummy myself but WCL treats it as a trash fight and so Analyzer doesn't even show it. Some LFR maybe. I'll look into it (same problem will be happening with Soul Harvest as it buffs pet's damage too)

File: src/Parser/AfflictionWarlock/Modules/WarlockCore/DeathsEmbrace.js (hunk 2 above)
Comment (sMteX) [inline src/Parser/AfflictionWarlock/Modules/WarlockCore/DeathsEmbrace.js]: I reworked this module in next commits. This mechanic is used by both the talent module and Soul of the Netherlord. I will be doing another PR this evening and this exact module is moved back into talents, and 2 new modules are created - DeathsEmbraceTalent and SoulOfTheNetherlord that both just get this module as a dependency and display the bonusDmg, but this time in its correct place (StatisticBox if it's from a talent, Item if it's from the legendary finger).
